openrtx/src/main.c

```diff
@@ -74,6 +77,17 @@ int main(void)
     // Create OpenRTX threads
     create_threads();
 
+#ifndef PLATFORM_LINUX
     // Jump to the UI task
     ui_task(NULL);
+#else
+    // macOS requires SDL main loop to run on the main thread.
+    // Here we create a new thread for ui_task and utilize the masin thread for
+    // the SDL main loop.
+    pthread_t      ui_thread;
+    pthread_create(&ui_thread, NULL, ui_task, NULL);
+
+    sdl_task();
```

Comment:
When I moved all the DSL logic into a custom thread I ended up with crashes at boot time. On macOS you can run the SDL main loop only on the Main Thread :(

So I had to rewire a bit of logic here in main, if on linux the `ui_task` runs on a thread and the `sdl_task` runs on the main thread.